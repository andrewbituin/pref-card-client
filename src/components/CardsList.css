.cards-list-navigation{
    background-color:rgb(50, 127, 191);
    color: white;
    padding: 20px;

}
.cards-list-header{
    margin: 6px;
}
.cards-list-title{
    margin: 0;
    font-size: 3em;
}
.cards-list{
    list-style: none;
}
ul{
    padding: 8px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
}
li{
    display: inline-block;
    width: 16%;
    border: 1px solid lightgray;
    margin: 5px;
    padding: 15px;
    border-radius: 15px;
    box-shadow: 5px 5px lightgray;
    background-color: white;
    text-align: center;
}
