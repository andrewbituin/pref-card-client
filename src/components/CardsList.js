import React from "react";
import { Link } from "react-router-dom";
import CardsCondensedList from "./CardsCondensedList";
import LogoutButton from "./LogoutButton";
import "./CardsList.css";

export default class CardsList extends React.Component {
  render() {
    return (
      <div>
        <nav className="cards-list-navigation">
          <header className="cards-list-header">
            <h1 className="cards-list-title">prefcards</h1>
          </header>
          <Link to="/create-card">
            <button type="click">New Card</button>
          </Link>
          <LogoutButton />
        </nav>
        <ul className="cards-list">
          <CardsCondensedList {...this.props} />
        </ul>
      </div>
    );
  }
}
