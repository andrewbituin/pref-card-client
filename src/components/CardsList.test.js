import React from "react";
import ReactDOM from "react-dom";
import CardsList from "./CardsList";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("CardsList component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <CardsList />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <CardsList />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});