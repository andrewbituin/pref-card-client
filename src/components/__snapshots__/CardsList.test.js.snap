// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList component renders the UI as expected 1`] = `
<div>
  <nav
    className="cards-list-navigation"
  >
    <header
      className="cards-list-header"
    >
      <h1
        className="cards-list-title"
      >
        prefcards
      </h1>
    </header>
    <a
      href="/create-card"
      onClick={[Function]}
    >
      <button
        type="click"
      >
        New Card
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        onClick={[Function]}
        type="click"
      >
        Logout
      </button>
    </a>
  </nav>
  <ul
    className="cards-list"
  />
</div>
`;
